### PR TITLE
Use ElemSideBuilder to avoid extraneous element side construction

### PR DESCRIPTION
**Pull request by loganharbour**
Refs #19605

Pending a libmesh update for https://github.com/libMesh/libmesh/pull/3100.

FYI @YaqiWang

### CODE REVIEW

File: framework/src/base/Assembly.C [hunk 1]
```diff
@@ -2165,8 +2154,6 @@ Assembly::computeADFace(const Elem * elem, unsigned int side)
 
   if (_subproblem.haveADObjects())
   {
-    const std::unique_ptr<const Elem> side_elem(elem->build_side_ptr(side));
```
Comment (loganharbour) [inline framework/src/base/Assembly.C:2168]: @lindsayad you say wat?!

Build an element side for _every_ AD face calc even without displaced mesh

Comment (lindsayad) [reply framework/src/base/Assembly.C:2168]: 👍

**Comment by loganharbour** [comment]
@lindsayad this is ready for a review

File: framework/src/base/Assembly.C [hunk 2]
```diff
@@ -2159,14 +2146,12 @@ Assembly::reinitElemFaceRef(const Elem * elem,
 }
 
 void
-Assembly::computeADFace(const Elem * elem, unsigned int side)
+Assembly::computeADFace(const Elem * elem, const unsigned int side)
```
Comment (lindsayad) [inline framework/src/base/Assembly.C:2149]: You changed the other method signature to a reference. Might as well do the same here. I believe this is only used internally in this class so shouldn't have to change any code outside of here. We dereference the pointer on the first line so a reference would be safer.

Comment (loganharbour) [reply framework/src/base/Assembly.C]: Sounds good, works for me. I've been slowly cleaning up internal signatures like this as I see them

**Review by lindsayad — APPROVED** [review]

**Pull request merged**